java-checks/src/main/java/org/sonar/java/checks/RedundantThrowsDeclarationCheck.java

```diff
@@ -224,18 +226,31 @@ private static boolean emptyBody(MethodTree methodTree) {
   private static class MethodInvocationVisitor extends BaseTreeVisitor {
     private Set<Type> thrownExceptions = new HashSet<>();
     private boolean visitedUnknown = false;
+    private boolean visitedOtherConstructor = false;
+    private final MethodTree methodTree;
+
+    MethodInvocationVisitor(MethodTree methodTree) {
+      this.methodTree = methodTree;
+    }
 
     @Nullable
     public Set<Type> thrownExceptions() {
       if (visitedUnknown || thrownExceptions.stream().anyMatch(Type::isUnknown)) {
         // as soon as there is an unknown type, we discard any attempt to find an issue
         return null;
       }
+      if (methodTree.is(Tree.Kind.CONSTRUCTOR) && !visitedOtherConstructor) {
+        Optional<Symbol.MethodSymbol> constructor = getImplicitlyCalledConstructor(methodTree);
+        constructor.ifPresent(c -> thrownExceptions.addAll(c.thrownTypes()));
+      }
       return thrownExceptions;
     }
 
     @Override
     public void visitMethodInvocation(MethodInvocationTree tree) {
+      if ("<init>".equals(tree.symbol().name())) {
```

Comment:
In my opinion, this is enough, if any `<init>` is called inside the body (this or super), the implicit constructor will not be present!